cmd/oslo/validate/validate.go

```diff
@@ -21,22 +21,31 @@ import (
 	"github.com/spf13/cobra"
 
 	"github.com/OpenSLO/oslo/internal/pkg/validate"
+	"github.com/OpenSLO/oslo/pkg/discoverfiles"
 )
 
 // NewValidateCmd returns a new cobra.Command for the validate command.
 func NewValidateCmd() *cobra.Command {
-	return &cobra.Command{
+	var passedFilePaths []string
+	var recursive bool
+
+	validateCmd := &cobra.Command{
 		Use:   "validate",
 		Short: "Validates your yaml file against the OpenSLO spec.",
 		Long:  `Validates your yaml file against the OpenSLO spec.`,
-		Args:  cobra.MinimumNArgs(1),
+		Args:  cobra.MinimumNArgs(0),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if e := validate.Files(args); e != nil {
-				return e
+			discoveredFilePaths, err := discoverfiles.DiscoverFilePaths(passedFilePaths, recursive)
+			if err != nil {
+				return err
+			}
+			if err := validate.Files(discoveredFilePaths); err != nil {
+				return err
 			}
-
 			fmt.Println("Valid!")
```

Comment:
I know this isn't scope of this issue but, maybe you could move this message from here, it doesn't tell to much to the user. I think better place would be to print some message in validate.Files. 

Something like that:

``` go
// Files validates the given array of filenames.
func Files(files []string) error {
	var allErrors []string
	for _, file := range files {
		c, e := yamlutils.ReadConf(file)
		if e != nil {
			allErrors = append(allErrors, e.Error())
			break
		}

		content, err := yamlutils.Parse(c, file)
		if err != nil {
			allErrors = append(allErrors, err.Error())
			break
		}
		if validationErrors := validateStruct(content); validationErrors != nil {
			fmt.Printf("%s validation error:\n%s\n\n", file, validationErrors.Error())
		}
		fmt.Printf("%s is valid", file)
	}
	if len(allErrors) > 0 {
		return errors.New(strings.Join(allErrors, "\n"))
	}
	return nil
}
```

I leave it up to you, I can suggest that change in different PR.

Reply:
I think we should create a separate issue to go through all messages and errors in `oslo` (I can do it) and adjust them, thus I prefer to leave it here as it is. What do you think?

Reply:
Makes sense